### PR TITLE
Accounts rest

### PR DESCRIPTION
**Pull request by nharraud**
* New users and roles rest api
* Updates Invenio dependencies.

This PR is waiting for inveniosoftware/invenio-accounts-rest#23 to be merged and the next alpha version of  inveniosoftware/invenio-accounts-rest to be released.

### CODE REVIEW

**Comment by nharraud** [comment]
The develop and current builds still fail because of inveniosoftware/invenio-oaiserver#112

File: b2share/modules/users/views.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
 blueprint = Blueprint(
     'b2share_users',
     __name__,
-    url_prefix='/users'
+    url_prefix='/user'
```
Comment (emanueldima) [inline b2share/modules/users/views.py]: This change breaks the UI, please update this file: https://github.com/EUDAT-B2SHARE/b2share/blob/evolution/webui/src/data/server.js#L14 (just the urls should be enough)

File: b2share/modules/communities/api.py [hunk 2]
```diff
@@ -217,7 +215,6 @@ def patch(self, patch):
             'description': self.model.description,
             'logo': self.model.logo,
             'publication_workflow': self.model.publication_workflow,
-            'restricted_submission': self.model.restricted_submission,
```
Comment (emanueldima) [inline b2share/modules/communities/api.py]: Is this change intended?

File: b2share/modules/communities/api.py [hunk 3]
```diff
@@ -139,12 +141,8 @@ def create_community(cls, name, description, logo=None, id_=None,
                 kwargs = {}
                 if id_ is not None:
                     kwargs['id'] = id_
-                model = CommunityMetadata(name=name,
-                                          description=description,
-                                          logo=logo,
-                                          publication_workflow=publication_workflow,
```
Comment (emanueldima) [inline b2share/modules/communities/api.py]: Is this change intended?

File: b2share/modules/communities/api.py [hunk 4]
```diff
@@ -95,19 +97,19 @@ def get_all(cls, start=None, stop=None, name=None):
         """Searches for matching communities."""
         if (start is None and stop is None):
             if name is None:
-                metadata = CommunityMetadata.query.order_by(CommunityMetadata.created)
+                metadata = CommunityMeta.query.order_by(CommunityMeta.created)
```
Comment (nharraud) [inline b2share/modules/communities/api.py]: Oups I think you are right. I have rebased and added this by mistake.

Comment (nharraud) [inline b2share/modules/communities/api.py]: Stange that the tests didn't detect it.

File: b2share/modules/users/views.py (hunk 1 above)
Comment (nharraud) [inline b2share/modules/users/views.py]: ok thanks !
